tutorials/inner_products/3_calculus.py

```diff
@@ -0,0 +1,265 @@
+
+# Items required to perform u.T*(Me*Cf*v)
+Me = mesh.getEdgeInnerProduct()  # Basic inner product matrix (edges)
+Cf = mesh.edgeCurl.T             # Faces to edges curl (assumes Dirichlet)
+    
+# Plot Sparse Representation
+fig = plt.figure(figsize=(9, 5))
+
+Ax1 = fig.add_subplot(121)
+Ax1.spy(Mf*Ce, markersize=0.5)
+Ax1.set_title('Mf*Ce', pad=10)
+
+Ax2 = fig.add_subplot(122)
+Ax2.spy(Me*Cf, markersize=0.5)
+Ax2.set_title('Me*Cf', pad=10)
+
+
+###########################################################
+# Scalar Laplacian
+# ----------------
+# 
+# Where :math:`\psi` and :math:`\phi` are scalar quantities, and the scalar
+# Laplacian :math:`\Delta^2 = \nabla \cdot \nabla`, we would like to
+# approximate the following inner product:
+#
+# .. math::
+#     (\psi , \nabla \cdot \nabla \phi) = \int_\Omega \psi (\nabla \cdot \nabla \phi) \, dv
+#
+# Using :math:`p \nabla \cdot \mathbf{q} = \nabla \cdot (p \mathbf{q}) - \mathbf{q} \cdot (\nabla p )`
+# and the Divergence theorem we obtain:
+#
+# .. math::
+#     \int_{\partial \Omega} \mathbf{n} \cdot ( \psi \nabla \phi ) \, da
+#     - \int_\Omega (\nabla \psi ) \cdot (\nabla \phi ) \, dv
+# 
+# In this case, the surface integral can be eliminated if we can assume a
+# Neumann condition of :math:`\partial \phi/\partial n = 0` on the boundary.
+#
+# **Inner Prodcut at Edges:**
+#
+# Let :math:`\psi` and :math:`\phi` be discretized to the nodes. In this case,
+# the discrete gradient operator (:math:`\mathbf{G_n}`) must map from nodes
+# to edges. Ultimately we evaluate the inner product using an inner product
+# matrix (:math:`\mathbf{M_e}` for quantities living on cell edges, e.g.:
+#
+# .. math::
+#     (\psi , \nabla \cdot \nabla \phi) \approx \mathbf{\psi G_n^T M_e G_n \phi}
+#
+# **Inner Product at Faces:**
+#
+# Let :math:`\psi` and :math:`\phi` be discretized to cell centers. In this
+# case, the discrete gradient operator (:math:`\mathbf{G_c}`) must map from
+# centers to faces; and requires the user to set Neumann conditions in the
+# operator. Ultimately we evaluate the inner product using an inner product
+# matrix (:math:`\mathbf{M_f}`) for quantities living on cell faces, e.g.:
+#
+# .. math::
+#     (\psi , \nabla \cdot \nabla \phi) \approx \mathbf{\psi G_c^T M_f G_c \phi}
+#
+# 
+
+# Make basic mesh
+h = np.ones(10)
+mesh = TensorMesh([h, h, h])
+
+# Items required to perform psi.T*(Gn.T*Me*Gn*phi)
+Me = mesh.getEdgeInnerProduct()  # Basic inner product matrix (edges)
+Gn = mesh.nodalGrad              # Nodes to edges gradient
+
+# Items required to perform psi.T*(Gc.T*Mf*Gc*phi)
+Mf = mesh.getFaceInnerProduct()  # Basic inner product matrix (faces)
+mesh.setCellGradBC(['dirichlet','dirichlet','dirichlet'])
+Gc = mesh.cellGrad            # Centers to faces gradient
+
+# Plot Sparse Representation
+fig = plt.figure(figsize=(9, 4))
+
+Ax1 = fig.add_subplot(121)
+Ax1.spy(Gn.T*Me*Gn, markersize=0.5)
+Ax1.set_title('Gn.T*Me*Gn', pad=5)
```

Comment:
A simple hack for not having the title overlap would be to add a new line (`Gn.T*Me*Gn\n`)